input/kafkamdm/lag_monitor.go

```diff
@@ -190,6 +174,40 @@ func (l *LagMonitor) Metric() int {
 	return max
 }
 
+func (l *LagMonitor) getPartitionPriority(partition int32, lag *lagLogger) Status {
+	status := Status{
+		Lag:  lag.Min(),       // accurate lag, -1 if unknown
+		Rate: int(lag.Rate()), // accurate rate, or 0 if we're not sure
+	}
+	if status.Lag == -1 {
+		// if we have no lag measurements yet,
+		// just assign a priority of 10k for this partition
+		status.Priority = 10000
+	} else {
+		// if we're not sure of rate, we don't want divide by zero
+		// instead assume rate is super low
+		if status.Rate == 0 {
+			status.Priority = status.Lag
+		} else {
+			status.Priority = status.Lag / status.Rate
+		}
+	}
+
+	return status
+}
+
+func (l *LagMonitor) GetPartitionPriority(partition int32) int {
+	var lag *lagLogger
```

Comment:
dont you need to acquire l.Lock() here?

Reply:
right, don't know how i missed that, will fix it